Training_25/Program.cs

```diff
@@ -3,11 +3,49 @@
 // Copyright (c) Metamation India.
 // ------------------------------------------------------------------
 // Program.cs
-// Program on main branch.
+// Program on T01 branch - Assignments Q1: NUMBER CONVERSION GAME
 // ------------------------------------------------------------------------------------------------
+using System.Text;
+using static System.Console;
+using static System.Math;
 namespace Training_25;
+
 internal class Program {
    static void Main (string[] args) {
-      Console.WriteLine ("Hello, World!");
+      WriteLine ("Enter a number");
+      string? input = ReadLine ();
+      int num;
+      while (string.IsNullOrWhiteSpace (input) || !int.TryParse (input, out num)) {
+         WriteLine ("Invalid input. Please enter a  number:");
+         input = ReadLine ();
+      }
+      WriteLine ($"Input: {num}");
+      int n = num;
+      if (n == 0) WriteLine ("HEX: 0");
+      else {
+         bool isNegative = n < 0;
+         n = Abs (n);
+         StringBuilder sb = new ();
+         while (n > 0) {
+            int r = n % 16;
+            n /= 16;
+            if (r < 10) sb.Insert (0, r);
+            else
+               sb.Insert (0, (char)('A' + r - 10));

```

Comment:
Replace if else loop with ternary operator.
```suggestion
               sb.Insert (0, r < 10 ? r : (char)('A' + r - 10));
```